### PR TITLE
Create Authentication module

### PR DESCRIPTION
**Pull request by gleybersonandrade**

### CODE REVIEW

**Review by ajoaoff — CHANGES_REQUESTED** [review]
The methods to manipulate the user database should themselves be used only by authenticated users. There should be a method to create an initial user on deploy, or through CLI on the server.

**Review by ajoaoff — CHANGES_REQUESTED** [review]
I believe `_list_users` and `_list_user` should also require an authenticated user.
The method to create the default user should not be called by `__init__` , because the default user will be recreated every time Kytos is started.

**Review by hdiogenes — APPROVED** [review]

File: kytos/core/auth.py [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+            if error or not box:
+                response = {
+                    "answer": "User data cannot be shown",
+                    "code": HTTPStatus.INTERNAL_SERVER_ERROR.value,
+                }
+            else:
+                response = {
+                    "answer": {"data": box.data},
+                    "code": HTTPStatus.OK.value,
+                }
```
Comment (ajoaoff) [inline kytos/core/auth.py]: An empty `box` is not an error, it means there is no user with the given uid.

Comment (ajoaoff) [inline kytos/core/auth.py]: An empty `box` is not an error, it means there is no user with the given uid.

File: kytos/core/auth.py [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+            result = f"Incorrect username or password: {exc}"
+            return result, HTTPStatus.UNAUTHORIZED.value
+
+    def _find_user(self, uid):
+        """Find a specific user using Storehouse."""
+        response = {}
+
+        def _find_user_callback(_event, box, error):
+            nonlocal response
+            if error or not box:
```
Comment (ajoaoff) [inline kytos/core/auth.py]:
```suggestion
            if error:
```

File: kytos/core/auth.py [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+                response = {
+                    "answer": "User data cannot be shown",
+                    "code": HTTPStatus.INTERNAL_SERVER_ERROR.value,
+                }
+            else:
```
Comment (ajoaoff) [inline kytos/core/auth.py:174]:
```suggestion
            elif not box:
                response = {
                    "answer": f'User with uid {uid} not found',
                    "code": HTTPStatus.NOT_FOUND.value
                }
            else:
```

File: kytos/core/auth.py [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+            if error or not box:
+                response = {
+                    "answer": "User has not been deleted",
+                    "code": HTTPStatus.INTERNAL_SERVER_ERROR.value,
+                }
+            else:
+                response = {
+                    "answer": "User successfully deleted",
+                    "code": HTTPStatus.OK.value,
+                }
```
Comment (ajoaoff) [inline kytos/core/auth.py]: Same as `_find_user`. Here maybe it would be better to check first if the user exists.

**Review by ajoaoff — CHANGES_REQUESTED** [review]

**Review by ajoaoff — APPROVED** [review]

File: kytos/core/config.py [hunk 5]
```diff
@@ -76,6 +76,10 @@ def __init__(self):
                             action='store_true',
                             help="Enable all new Entities by default.")
 
+        parser.add_argument('-C', '--create_superuser',
```
Comment (hdiogenes) [inline kytos/core/config.py]:
```suggestion
        parser.add_argument('-C', '--create-superuser',
```

**Pull request merged**